spec/app/domain/authentication/authn/authenticator_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe 'Authentication::Authn::Authenticator' do
+  include_context "authn setup"
+
+  context "An Authn authenticator" do
+    context "that receives an authentication request" do
+      context "with valid credentials" do
+        subject do
+           ::Authentication::Authn::Authenticator.new(
+                role_cls:        mocked_role,
+                credentials_cls: mocked_valid_api_key_credentials
```

Comment:
Minorish point: I would remove the `mocked_` prefix from all of these variable names.   I think it's implied by the context and obscures the actual meaning that we care about, which is what comes after the prefix.   It's also just harder to parse long names visually.